### PR TITLE
fix: hold wallet shared pointer in CJ Manager/Sessions to prevent concurrent unload

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
https://github.com/dashpay/dash/pull/6440#discussion_r1865042366

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by knst — CHANGES_REQUESTED** [review]
NACK this solution is over-complex compare to previous approach.

See my clarification there: https://github.com/dashpay/dash/pull/6440#discussion_r1867193382

File: src/coinjoin/client.h [hunk 1]
```diff
@@ -138,7 +138,7 @@ class CoinJoinWalletManager {
 class CCoinJoinClientSession : public CCoinJoinBaseSession
 {
 private:
-    CWallet& m_wallet;
+    const std::shared_ptr<CWallet> m_wallet;
```
Comment (knst) [inline src/coinjoin/client.h:141]: Alternatively,
if CCoinJoinClientSession and CCoinJoinClientManager is now an owner of CWallet, smart pointer still can be removed from `CTransactionBuilder`.

Comment (knst) [reply src/coinjoin/client.h:141]: so, with this PR we actually have guarantee that CWallet won't be freed during any CCoinJoinClientSession and CCoinJoinClientManager operation. Let's do a final touch then: 4f731914941564fcd41972e33dab92e5b0d8e011 to simplify code a bit further

**Review by PastaPastaPasta — APPROVED** [review]
utACK 2d7c7f81b8a4e2b31986129bcc76fc012a324a26

**Pull request merged**